src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp

```diff
@@ -1481,23 +1483,30 @@ bool DiscoveryDataBase::process_dirty_topics()
                         // Check the status of the writer in `readers_[reader]::relevant_participants_builtin_ack_status`.
```

Comment:
I think that we can move here the `readers_it = readers_.find(reader)` and prevent the `find()` if participants are not matched